src/cpp/rtps/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -310,7 +310,7 @@ TCPTransactionId RTCPMessageManager::sendConnectionRequest(
     }
     else
     {
-        IPLocator::setPhysicalPort(locator, static_cast<uint16_t>(SystemInfo::instance().process_id()));
+        IPLocator::setPhysicalPort(locator, IPLocator::getPhysicalPort(mTransport->get_initial_peer_local_locator()));
```

Comment:
Instead of `get_initial_peer_local_locator` it would be better to move the code in lines 306-314 into a `fill_local_physical_port(locator)` method on `TCPTransportInterface`. This way we hide the implementation details.